networkx/algorithms/distance_measures.py

```diff
@@ -412,6 +412,26 @@ def harmonic_diameter(G, sp=None):
     G : NetworkX graph
        A graph
 
+    weight : string, function, or None (default=None)
+       If this is a string, then edge weights will be accessed via the
+       edge attribute with this key (that is, the weight of the edge
+       joining `u` to `v` will be ``G.edges[u, v][weight]``). If no
+       such edge attribute exists, the weight of the edge is assumed to
+       be one.
+
+    If this is a function, the weight of an edge is the value
+    returned by the function. The function must accept exactly three
+    positional arguments: the two endpoints of an edge and the
+    dictionary of edge attributes for that edge. The function must
+    return a number.
+
+    If this is None, every edge has weight/distance/cost 1.
+
+    Weights stored as floating point values can lead to small round-off
+    errors in distances. Use integer weights to avoid this.
+
+    Weights should be positive, since they are distances.
```

Comment:
Can you use the doc_string language for `weight` from one of the existing functoins that allows `weight` to be a function? That provides consistency and eases review.

Reply:
1. Updated the `weight` parameter description to match the style used in other NetworkX functions.

2. Moved the notes about floating-point weights and positive weights to a separate "Notes" section at the end of the docstring.

3. Kept the rest of the docstring, including the detailed explanation and references, unchanged.